bindings/redis/redis.go

```diff
@@ -66,7 +66,7 @@ func (r *Redis) Init(meta bindings.Metadata) error {
 	}
 
 	r.client = redis.NewClient(opts)
-	_, err = r.client.Ping().Result()
+	_, err = r.client.Ping(context.Background()).Result()
```

Comment:
I think it makes sense to add a `ctx` context variable in the `Redis` struct to handle cancelling the child contexts when closing the component. `r.ctx` would replace anywhere `context.Background()` is used. See the pubsub component.

Reply:
I've add ctx and cancel the same as the pubsub component.Also implement `Close() error` function.